File: examples/governance_static_go/domain/submit_assessment.go
package domain

import (
	"time"
	"fmt"
)

type SubmitAssessment struct {
	AssessmentId string `json:"assessment_id"`
	RiskLevel string `json:"risk_level"`
	BiasScore float64 `json:"bias_score"`
	SafetyScore float64 `json:"safety_score"`
	TransparencyScore float64 `json:"transparency_score"`
	OverallScore float64 `json:"overall_score"`
}

func (c SubmitAssessment) CommandName() string { return "SubmitAssessment" }

func (c SubmitAssessment) Execute(repo AssessmentRepository) (*Assessment, *SubmittedAssessment, error) {
	existing, err := repo.Find(c.AssessmentId)
	if err != nil {
		return nil, nil, err
	}
	if existing == nil {
		return nil, nil, fmt.Errorf("Assessment not found: %s", c.AssessmentId)
	}
	existing.RiskLevel = c.RiskLevel
	existing.BiasScore = c.BiasScore
	existing.SafetyScore = c.SafetyScore
	existing.TransparencyScore = c.TransparencyScore
	existing.OverallScore = c.OverallScore
	existing.UpdatedAt = time.Now()
	if err := existing.Validate(); err != nil {
		return nil, nil, err
	}
	if err := repo.Save(existing); err != nil {
		return nil, nil, err
	}
	event := SubmittedAssessment{
		AggregateID: existing.ID,
		AssessmentId: c.AssessmentId,
		RiskLevel: c.RiskLevel,
		BiasScore: c.BiasScore,
		SafetyScore: c.SafetyScore,
		TransparencyScore: c.TransparencyScore,
		OverallScore: c.OverallScore,
		OccurredAt: time.Now(),
	}
	return existing, &event, nil
}

File: examples/governance/risk_assessment_domain/lib/risk_assessment_domain/assessment/commands/submit_assessment.rb
module RiskAssessmentDomain
  class Assessment
    module Commands
      class SubmitAssessment
        include Hecks::Command
        emits "SubmittedAssessment"

        attr_reader :assessment_id
        attr_reader :risk_level
        attr_reader :bias_score
        attr_reader :safety_score
        attr_reader :transparency_score
        attr_reader :overall_score

        def initialize(
          assessment_id: nil,
          risk_level: nil,
          bias_score: nil,
          safety_score: nil,
          transparency_score: nil,
          overall_score: nil
        )
          @assessment_id = assessment_id
          @risk_level = risk_level
          @bias_score = bias_score
          @safety_score = safety_score
          @transparency_score = transparency_score
          @overall_score = overall_score
        end

        def call
          existing = repository.find(assessment_id)
          if existing
            unless existing.status == "pending"
              raise Hecks::Error, "Cannot SubmitAssessment: status must be 'pending', got '#{existing.status}'"
            end
            Assessment.new(
              id: existing.id,
              model_id: existing.model_id,
              assessor_id: existing.assessor_id,
              risk_level: risk_level,
              bias_score: bias_score,
              safety_score: safety_score,
              transparency_score: transparency_score,
              overall_score: overall_score,
              findings: existing.findings,
              mitigations: existing.mitigations,
              submitted_at: Time.now.to_s,
              status: "submitted"
            )
          else
            raise Hecks::Error, "Assessment not found: #{assessment_id}"
          end
        end
      end
    end
  end
end
